protobuf/echo/EchoInstantiation.hpp

```diff
@@ -68,6 +73,11 @@ namespace main_
             , to(toSerial)
         {}
 
+        ~EchoForwarderToSerial()
+        {
+            this->RemoveAllForwarders();
```

Comment:
Why not just a forwarders.clear() here?

Reply:
Indeed; I don't see why someone outside of a destruction situation would want to remove all forwarders in one go

Reply:
Because I didn't want to increase the scope of the forwarders object outside of the EchoForwarder class. Is it an idea to  make the RemoveAllForwarders method protected instead?

Reply:
Aah, I missed a bit of context in the diff.

In that case, why not add a destructor to the base class to clean-up its resources.

Reply:
After discussion, I have now made the echoForwarder an object of EchoForwarderToSerial instead of inheriting from it. This resolves the problem with order of destruction in a cleaner way.